addons/controllers/cpi/vspherecpiconfig_controller.go

```diff
@@ -175,8 +175,9 @@ func (r *VSphereCPIConfigReconciler) reconcileVSphereCPIConfigNormal(ctx context
 	if *cpiConfig.Spec.VSphereCPI.Mode == VSphereCPIParavirtualMode {
 		// create an aggregated cluster role RBAC that will be inherited by CAPV (https://kubernetes.io/docs/reference/access-authn-authz/rbac/#aggregated-clusterroles)
 		// CAPV needs to hold these rules before it can grant it to serviceAccount for CPI
+
 		_, err := controllerutil.CreateOrPatch(ctx, r.Client, constants.CAPVAggregatedClusterRole, func() error {
-			constants.CAPVAggregatedClusterRole.Rules = providerServiceAccountRBACRules
+			constants.CAPVAggregatedClusterRole.Rules = append(constants.CAPVAggregatedClusterRole.Rules, providerServiceAccountRBACRules...)
```

Comment:
dumb question: Since we are using **constants**.CAPVAggregatedClusterRole doesnt it mutate this var everytime the controller runs ?

i.e. first run it adds providerServiceAccountRBACRules and second time this run doesnt it append again ?

Reply:
yup it does but controllerutil.CreateOrPatch does the diff before applying